core/src/main/java/com/novoda/noplayer/exoplayer/ExoPlayerFacade.java

```diff
@@ -104,18 +115,55 @@ public void loadVideo(Uri uri, ContentType contentType, ExoPlayerForwarder forwa
                 forwarder.mediaSourceEventListener()
         );
         exoPlayer.prepare(mediaSource, RESET_POSITION, DO_NOT_RESET_STATE);
+        setExoPlayerTextOutput(output);
     }
 
     public void selectAudioTrack(PlayerAudioTrack audioTrack) {
-        // TODO check if we can read tracks from exoplayer directly
-        trackSelector.selectAudioTrack(audioTrack);
+        audioTrackSelector.selectAudioTrack(audioTrack, rendererTypeRequester);
     }
 
     public List<PlayerAudioTrack> getAudioTracks() {
-        return trackSelector.getAudioTracks();
+        return audioTrackSelector.getAudioTracks(rendererTypeRequester);
+    }
+
+    public void setSubtitleRendererOutput(TextRenderer.Output output) {
+        this.output = output;
+        setExoPlayerTextOutput(output);
+    }
+
+    private void setExoPlayerTextOutput(TextRenderer.Output output) {
+        if (exoPlayer == null) {
+            return;
+        }
+
+        exoPlayer.setTextOutput(output);
+    }
+
+    public void selectSubtitleTrack(PlayerSubtitleTrack subtitleTrack) {
+        subtitleTrackSelector.selectTextTrack(subtitleTrack, rendererTypeRequester);
+    }
+
+    public List<PlayerSubtitleTrack> getSubtitleTracks() {
+        return subtitleTrackSelector.getSubtitleTracks(rendererTypeRequester);
     }
 
     public boolean hasPlayedContent() {
         return exoPlayer != null;
     }
+
+    public void clearSubtitleTrack() {
+        subtitleTrackSelector.clearSubtitleTrack(rendererTypeRequester);
+    }
+
+    public void selectFirstAvailableSubtitlesTrack() {
+        subtitleTrackSelector.selectFirstTextTrack(rendererTypeRequester);
+    }
+
+    @VisibleForTesting
+    final RendererTypeRequester rendererTypeRequester = new RendererTypeRequester() {
```

Comment:
You can create a `mock` of the `ExoPlayer` which means you can control the call to `exoPlayer.getRendererType(index)`.